src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs

```diff
@@ -133,6 +124,29 @@ protected override void ProcessRecord()
             }
         }
 
+        /// <summary>
+        /// Calculate the hash data from the Stream with the specified hash algorithm
+        /// </summary>
+        /// <returns>The computed hash data</returns>
+        private byte[] ComputeHash(Stream stream)
+        {
+            switch (Algorithm)
+            {
+                case HashAlgorithmNames.SHA1:
+                    return SHA1.HashData(stream);
+                case HashAlgorithmNames.SHA256:
+                    return SHA256.HashData(stream);
+                case HashAlgorithmNames.SHA384:
+                    return SHA384.HashData(stream);
+                case HashAlgorithmNames.SHA512:
+                    return SHA512.HashData(stream);
+                case HashAlgorithmNames.MD5:
+                    return MD5.HashData(stream);
+                default:
+                    return SHA256.HashData(stream);
```

Comment:
Why do we return SHA256 instead of throwing `NotSupportedException`?

Reply:
I think we should use Debug.Assert here and remove UtilityCommonStrings.AlgorithmTypeNotSupported from resx file.

Reply:
Done